File: auth-plugin/app/auth/interfaces/IAuthCache.java
package auth.interfaces;

import auth.models.AuthenticatedUser;
import auth.implementors.PlayAuthCache;
import com.google.inject.ImplementedBy;

@ImplementedBy(PlayAuthCache.class)
public interface IAuthCache {
    boolean isUserInCache(String session);
    String storeUserInCache(AuthenticatedUser authenticatedUser);
    AuthenticatedUser getUserFromCache(String session);
    boolean removeUserFromCache(String session);
    boolean refreshUserAuthenticationSession(String session);
}

File: auth-plugin/app/auth/interfaces/IAuthCache.scala
package auth.interfaces

import auth.models.AuthenticatedUser
import auth.implementors.PlayAuthCache
import com.google.inject.ImplementedBy

@ImplementedBy(classOf[PlayAuthCache])
trait IAuthCache {
    def isUserInCache(session: String): Boolean
    def storeUserInCache(authenticatedUser: AuthenticatedUser): String
    def getUserFromCache(session: String): AuthenticatedUser
    def removeUserFromCache(session: String): Boolean
    def refreshUserAuthenticationSession(session: String): Boolean
}
